### DISCUSSION TITLE
is there an alternative of ethers's contract.populateTransaction.METHOD_NAME

### DISCUSSION THREAD

**Discussion by Surya-Git-hub** [wevm/viem#688] [Question] [answered]
Hi, is there an alternative of ethers's contract.populateTransaction.METHOD_NAME in viem, if not how to create calldata or raw transaction to send as calldata using viem? please help!!

**Comment by jxom** [comment, marked as answer]
You can construct calldata with `encodeFunctionData`, then send that as `data` in `sendTransaction`.

**Reply by Surya-Git-hub** [reply]
thanks it will work

**Reply by joaoh9** [reply]
did it work? @Surya-Git-hub 
can you share a code snippet? I'm having the same problem

**Reply by Surya-Git-hub** [reply]
> did it work? @Surya-Git-hub can you share a code snippet? I'm having the same problem

yes it worked , i am using wagmi hooks to write transactions.
To generate calldata i am using encodeFunctionData like this
const callData = Object.keys(BasicOrderParameters)?.length ? encodeFunctionData({
            abi: seaportExchangeAbi,
            functionName: 'fulfillBasicOrder',
            args: [BasicOrderParameters]
        }) : "";

**Reply by keriat** [reply]
@jxom I see the view ```data``` optional argument but can I pass it to the [usePrepareSendTransaction](https://wagmi.sh/react/prepare-hooks/usePrepareSendTransaction) and [useSendTransaction](https://wagmi.sh/react/hooks/useSendTransaction)? The documentation isn't specifying this arg.
